cancelable/closer_test.go

```diff
@@ -0,0 +1,34 @@
+package cancelable
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testCloser struct {
+	closed bool
```

Comment:
i'd add an error field to this and return that in `Close()`. Then you can test that you are returning the error.

Reply:
Done